Rubberduck.VBEEditor/Extensions/VbeExtensions.cs

```diff
@@ -53,6 +54,36 @@ public static IHostApplication HostApplication(this VBE vbe)
         {
             if (vbe.ActiveVBProject == null)
             {
+                const int ctl_view_host = 106;
+
+                CommandBarControl host_app_control = vbe.CommandBars.FindControl(MsoControlType.msoControlButton, ctl_view_host);
+
+                if (host_app_control == null)
+                {
+                    return null;
+                }
+                else
+                {
+                    switch (host_app_control.Caption)
+                    {
+                        case "Microsoft Excel":
+                            return new ExcelApp();
+                        case "Microsoft Access":
+                            return new AccessApp();
+                        case "Microsoft Word":
+                            return new WordApp();
+                        case "Microsoft PowerPoint":
+                            return new PowerPointApp();
+                        case "Microsoft Outlook":
+                            return new OutlookApp();
+                        case "Microsoft Publisher":
+                            return new PublisherApp();
+                        case "AutoCAD":
+                            return null; //TODO - Confirm the button caption
```

Comment:
Perhaps use `string.Contains` instead? why not `return new AutoCADApp();` anyway?